lumigator/infra/mzai/helm/lumigator/templates/_helpers.tpl

```diff
@@ -49,3 +49,8 @@ Selector labels
 app.kubernetes.io/name: {{ include "lumigator.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
+
+{{- define "lumigator.consts" -}}
```

Comment:
This is an interesting approach, but I think it might make sense to use a 'regular' helper variable here

Reply:
We could also use `envFrom` and avoid having to set this variable in two places https://gist.github.com/troyharvey/4506472732157221e04c6b15e3b3f094

Reply:
The issue with regular variables is that the value will need to be defined in each template, since variables don't cross the `include` scope afaict. This ensures that the same value is used in both templates (gen_secret and deployment). I'm using a special root property here because I don't want values from Values to leak into our constants.

Another trick I tend to use instead of setting the values manually is doing a `fromJson` or `fromYaml` for constants and then assigning that to the `.Consts` property, although that one is trickier for the new reader.

Reply:
If you mean using a define as, well, as a define, _my_ issue with the define/include mechanism in Helm is that it mixes dealing with values and dealing with format. In this way the value (with its proper type and structure, in case of maps for example) is set in one place, and then the receiver can decide how to best format this. Including simple snippets is ok with the base mechanism, but for example if we want to start adding constants or "scoped" properties (`mistral` as a map with subproperties) it gets awkward with even the second or third property.

If you think a plain define is clearer in this case, I'll replace it.